resource-types/GithubStatus.dhall

```diff
@@ -0,0 +1,52 @@
+{ meta = { name = "github-status", repository = "resource/github-status" }
+, Version = { Type = {}, default = {=} }
+, Params =
+  { Get =
+    { Type = { path : Text, context : Optional Text }
+    , default.context = None Text
+    }
+  , Put =
+      let StatusObject =
+            { Type =
+                { state : Text
+                , context : Optional Text
+                , description : Optional Text
+                , target_url : Optional Text
+                }
+            , default =
+              { context = None Text
+              , description = None Text
+              , target_url = None Text
+              }
+            }
+
+      let Statuses =
+            { Type = { path : Text, statuses : List StatusObject.Type }
+            , default = {=}
+            }
+
+      let Status =
+            { Type = { path : Text } ⩓ StatusObject.Type
+            , default =
+              { context = None Text
+              , description = None Text
+              , target_url = None Text
+              }
+            }
+
+      let Put =
+              { Type = < Status : Status.Type | Statuses : Statuses.Type > }
+            ∧ { StatusObject, Status, Statuses }
+
+      let test =
+            Put.Type.Status Put.Status::{ path = "test", state = "successs" }
```

Comment:
My preferred idiom here is the one I used for `ConcoursePipeline`: https://github.com/coralogix/dhall-concourse/blob/491cbd1d01092997a264f960bea6c32fce37921c/resource-types/ConcoursePipeline.dhall#L46

```
let Put = < Status : Status.Type | Statuses : Statuses.Type >
in 
{ Type = Put,
  Status = \(status : Status.Type) -> Put.Status status ,
  Statuses = \(statuses : Statuses.Type) -> Put.Statuses statuses,
  Options = { Status, Statuses }
}
```

then the test is `Put.Status Put.Options.Status::{ path = "test" , state = "success" }`